lib/routes/toranoana/news.ts

```diff
@@ -0,0 +1,117 @@
+    categories: ['anime'],
+    example: '/toranoana/news/toragen',
+    parameters: { category: 'category' },
+    features: {
+        requireConfig: false,
+        requirePuppeteer: false,
+        antiCrawler: false,
+        supportBT: false,
+        supportPodcast: false,
+        supportScihub: false,
+    },
+    name: 'Category',
+    maintainers: ['Tsuyumi25'],
+    handler,
+    radar: [
+        {
+            title: '総合新着記事',
+            source: ['news.toranoana.jp'],
+            target: '/news',
+        },
+        {
+            title: '女性向け',
+            source: ['news.toranoana.jp/joshi'],
+            target: '/news/joshi',
+        },
+        {
+            title: 'イラスト展',
+            source: ['news.toranoana.jp/exhibitions'],
+            target: '/news/exhibition',
+        },
+        {
+            source: ['news.toranoana.jp/category/:category'],
+            target: '/news/:category',
+        },
+    ],
+    description: `
+::: warning TIP
+[総合新着記事](https://news.toranoana.jp)→\`/toranoana/news\`  
+[女性向け](https://news.toranoana.jp/joshi)→\`/toranoana/news/joshi\`  
+[イラスト展](https://news.toranoana.jp/exhibitions)→\`/toranoana/news/exhibition\`  
+[\`https://news.toranoana.jp/category/media\`](https://news.toranoana.jp/category/media)→\`/toranoana/news/media\`
+:::`,
+};
+
+async function handler(ctx): Promise<Data> {
+    const { category = '' } = ctx.req.param();
+    let apiUrl = 'https://news.toranoana.jp/wp-json/wp/v2/posts';
+
+    if (category) {
+        const categoryResponse = await ofetch(`https://news.toranoana.jp/wp-json/wp/v2/categories?slug=${category}`);
+        if (categoryResponse && categoryResponse.length > 0) {
+            apiUrl += `?categories=${categoryResponse[0].id}`;
+        }
+    } else {
+        // exclude category-joshi to get result of general
+        apiUrl += `?categories_exclude=1598`;
+    }
+
+    const posts = await ofetch(apiUrl, {
+        params: {
+            per_page: 20,
+        },
+    });
+
+    if (!posts || !posts.length) {
+        throw new Error('No posts found');
+    }
+
+    const items = await Promise.all(
+        posts.map(async (post) => {
+            const $ = load(post.content.rendered);
+
+            // remove unnecessary title
+            $('h1').first().remove();
+            $('h2').first().remove();
+
+            let thumbnail = '';
+            if (post.featured_media) {
+                try {
+                    const mediaData = await ofetch(`https://news.toranoana.jp/wp-json/wp/v2/media/${post.featured_media}`);
+                    thumbnail = mediaData.media_details?.sizes?.medium?.source_url || mediaData.source_url;
+                } catch (error) {
+                    logger.warn(`Failed to fetch media for ID ${post.featured_media}:`, error);
+                }
+            }
+
+            if (thumbnail) {
+                $('body').prepend(`<img src="${thumbnail}" alt="${post.title.rendered}" />`);
+            }
```

Comment:
featured media will be included in the `/post` API response if it's called with `_embed` search parameter